prov/shm/src/smr_rma.c

```diff
@@ -69,16 +72,27 @@ static ssize_t smr_rma_fast(struct smr_region *peer_smr, const struct iovec *iov
 	if (ret == -FI_ENOENT)
 		return -FI_EAGAIN;
 
-	memcpy(cma_iovec, iov, sizeof(*iov) * iov_count);
+	memcpy(vma_iovec, iov, sizeof(*iov) * iov_count);
 	for (i = 0; i < rma_count; i++) {
 		rma_iovec[i].iov_base = (void *) rma_iov[i].addr;
 		rma_iovec[i].iov_len = rma_iov[i].len;
 	}
 
 	total_len = ofi_total_iov_len(iov, iov_count);
 
-	ret = smr_cma_loop(peer_smr->pid, cma_iovec, iov_count, rma_iovec,
-			   rma_count, 0, total_len, op == ofi_op_write);
+	xpmem = &smr_peer_data(ep->region)[peer_id].xpmem;
```

Comment:
I believe this peer id is the one for the remote peer, not the local side. you'll need to pass in id in addition

Reply:
I'm not sure I follow the comment. I want to get the xpmem structure for the remote peer. Can you clarify what you mean by "pass in id in addition"? I don't think I need the local id in this code path.

Reply:
There are two "ids" - "id" is used to access peers in the local shm and "peer_id" is what the remote peer uses to access the sender in the remote shm. The peer id is only really used to save into the cmd so that the receiver can easily look up who the message is coming from. Whenever you're accessing the resources locally on the sender's shm (ie ep->region) you need to use the local id - ie "id"
See how smr_generic_rma uses the id

Reply:
i see what you mean now. thanks.